File: 75InterviewProblems/DynamicProgramming/ClimbingStairs.js
const climbStairs = (n) => {
    if (n <= 2) return n;
    let ways = [];
    ways[1] = 1;
    ways[2] = 2;

    for (let i = 3; i <= n; i++) {
        ways[i] = ways[i - 1] + ways[i - 2];
    }
    return ways[n]
}

File: 75InterviewProblems/DynamicProgramming/Python/ClimbingStairs.py
class Solution:
    def climbStairs(self, num: int) -> int:
        if num <= 2:
            return num
        
        ways = [1,2]

        print(ways)
        for i in range(2,num):
            ways.append(ways[i-1] + ways[i-2])
        return ways[len(ways)-1]


s = Solution()
print(s.climbStairs(4))